crates/sui-core/src/unit_tests/move_package_publish_tests.rs

```diff
@@ -422,3 +426,71 @@ async fn test_publish_extraneous_bytes_modules() {
         }
     )
 }
+
+#[tokio::test]
+#[cfg_attr(msim, ignore)]
+async fn test_publish_max_packages() {
+    let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+    let gas_object_id = ObjectID::random();
+    let authority = init_state_with_ids(vec![(sender, gas_object_id)]).await;
+
+    let (_, modules, dependencies) = build_package("object_basics", false);
+
+    // push max number of packages allowed to publish
+    let max_pub_cmd = authority
+        .epoch_store_for_testing()
+        .protocol_config()
+        .max_publish_or_upgrade_per_ptb_as_option()
+        .unwrap_or(0);
+    let mut packages = vec![];
+    for _ in 1..max_pub_cmd {
+        packages.push((modules.clone(), dependencies.clone()));
+    }
+    packages.push((modules, dependencies));
```

Comment:
I was initially a little confused by this pattern (pushing `N - 1` in the loop, and then the final `1` at the end), how about this instead?

```suggestion
    let packages = vec![(modules, dependencies); max_pub_cmd];
```